plenum/test/common/test_parse_ledger.py

```diff
@@ -25,14 +27,53 @@ def tdirWithLedger(tdir):
             txn[TARGET_NYM] = "invalid===="
         ledger.add(txn)
     return ledger
+
+@pytest.fixture(scope="module")
+def invalid_identifier(tdir):
+    tree = CompactMerkleTree()
+    ledger = Ledger(CompactMerkleTree(), dataDir=tdir)
+    txn = { TXN_TYPE: '0',
+            TARGET_NYM: base58.b58encode(b'whatever'),
+            IDENTIFIER: "invalid====",
+            DATA: {
+            NAME: str(2),
+            ALIAS: 'test' + str(2),
+            SERVICES: [VALIDATOR],
+            }
+          }
+    ledger.add(txn)
+    return ledger
+
+
+class DummyLedger(Ledger):
+    def getAllTxn(self, frm: int=None, to: int=None):
+        raise JSONDecodeError('', '', 0)
+
                                                             
 """
 Test that invalid base58 TARGET_NYM in pool_transaction raises the proper exception (INDY-150)
 """
 
-def test_parse_non_base58_txn_type_field_raises_descriptive_error(tdirWithLedger,tdir):
-    with pytest.raises(ValueError) as excinfo:
+def test_parse_verkey_non_base58_txn_type_field_raises_descriptive_error(tdirWithLedger,tdir):
+    with pytest.raises(SystemExit) as excinfo:
+        ledger = Ledger(CompactMerkleTree(), dataDir=tdir)
+        _, _, nodeKeys = TxnStackManager.parseLedgerForHaAndKeys(ledger)
+    assert excinfo.value.code == -1
+    ledger.stop()
+
+
+def test_parse_identifier_non_base58_txn_type_field_raises_descriptive_error(invalid_identifier, tdir):
+    with pytest.raises(SystemExit) as excinfo:
         ledger = Ledger(CompactMerkleTree(), dataDir=tdir)
         _, _, nodeKeys = TxnStackManager.parseLedgerForHaAndKeys(ledger)
-    assert("verkey" in str(excinfo.value))
+    assert excinfo.value.code == -1
     ledger.stop()
+
+
+def test_pool_file_is_invalid_json_format_raises_JSONDecodeError(tdir):
```

Comment:
Looks like the test method name is not fully descriptive: we just check that we're raising SystemExit.